tests/test_workers/test_pkg_managers/test_pip.py

```diff
@@ -654,3 +654,461 @@ def test_get_version_attr_with_package_dir(
         self._test_version_with_file_tree(
             project_tree, expect_version, expect_logs, expect_error, tmpdir, caplog
         )
+
+
+class TestSetupPY:
+    """SetupPY tests."""
+
+    @pytest.mark.parametrize("exists", [True, False])
+    def test_exists(self, exists, tmpdir):
+        """Test file existence check."""
+        if exists:
+            tmpdir.join("setup.py").write("")
+
+        setup_py = pip.SetupPY(tmpdir.strpath)
+        assert setup_py.exists() == exists
+
+    def _test_get_value(self, tmpdir, caplog, script_content, expect_val, expect_logs, what="name"):
+        """Test getting name or version from setup.py."""
+        tmpdir.join("setup.py").write(script_content.format(what=what))
+        setup_py = pip.SetupPY(tmpdir.strpath)
+
+        if what == "name":
+            value = setup_py.get_name()
+        elif what == "version":
+            value = setup_py.get_version()
+        else:
+            assert False, "'what' must be one of 'name', 'version'"
+
+        assert value == expect_val
+
+        logs = expect_logs.copy()
+        # Does not actually have to be at index 0, this is just to be more obvious
+        logs.insert(0, f"Parsing setup.py at '{tmpdir.join('setup.py')}'")
+        if expect_val is None:
+            logs.append(f"No {what} in setup.py, failed to resolve or invalid")
```

Comment:
@mprahl Would your recommendation to use the different style formatting for logs instead of f strings apply here as well?

Reply:
It wouldn't apply here since the code is comparing the already formatted log messages captured by pytest in `caplog` and not what was input to the respective log method.